sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/security/validation/CustomObjectMapperAdvice.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.spring.boot.web.security.validation;
+
+import javax.annotation.PostConstruct;
+import org.sdase.commons.spring.boot.web.jackson.SdaObjectMapperConfiguration;
+import org.sdase.commons.spring.boot.web.security.exception.InsecureConfigurationException;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.stereotype.Component;
+
+/**
+ * Checks that custom error mappers are registered by the SdaObjectMapperConfiguration. The check is
+ * indirectly performed by checking that the required bean is in the spring context. This class
+ * checks for the risks identified in the security guide as:
+ *
+ * <ul>
+ *   <li>Detection of confidential components
+ * </ul>
+ */
+@Component
+@ConditionalOnMissingBean(name = SdaObjectMapperConfiguration.OBJECT_MAPPER_BUILDER_BEAN_NAME)
+public class CustomObjectMapperAdvice {
+
+  private static final String JACKSON_CONFIGURATION_BEAN_CLASS = "Jackson2ObjectMapperBuilder";
+
+  @PostConstruct
+  public void check() {
+    throw new InsecureConfigurationException(
```

Comment:
Will we use the exception in more cases? At the moment the name sounds really generic but is only used in one very specific case. We could:
* use a better name
* use another `RuntimeException` (`IllegalStateException`?)

Reply:
Yes, there will be more insecure configuration to fail. And even if not, I think „insecure configuration“ is more meaningful than a generic Java exception.